src/sage/graphs/orientations.py

```diff
@@ -41,6 +41,233 @@
+        54
+
+    For C100 graph, we get first 10 orientations pretty quickly::
+
+        sage: from itertools import islice
+        sage: import timeit
+        sage: G_C100, start_time = graphs.CycleGraph(100), timeit.default_timer()
+        sage: it_C100, first_10_orientations = acyclic_orientations(G_C100), list(islice(acyclic_orientations(G_C100), 10))
+        sage: print(timeit.default_timer() - start_time < 0.05)
+        True
+
+    TESTS:
+
+    Acyclic orientations of a complete graph::
+
+        sage: G = graphs.CompleteGraph(5)
+        sage: it = g.acyclic_orientations(G)
+        sage: len(list(it))
+        120
+    """
+    from sage.rings.infinity import Infinity
+    from sage.combinat.subset import Subsets
+
+    def reorder_vertices(G):
+        n = G.order()
+        ko = n
+        k = n
+        G_copy = G.copy()
+        vertex_labels = [None] * n
+
+        while G_copy.size() > 0:
+            min_val = float('inf')
+            uv = None
+            for u, v, _ in G_copy.edges():
+                du = G_copy.degree(u)
+                dv = G_copy.degree(v)
+                val = (du + dv) / (du * dv)
+                if val < min_val:
+                    min_val = val
+                    uv = (u, v)
+
+            if uv:
+                u, v = uv
+                vertex_labels[u] = ko
+                vertex_labels[v] = ko - 1
+                G_copy.delete_vertex(u)
+                G_copy.delete_vertex(v)
+                ko -= 2
+
+            if G_copy.size() == 0:
+                break
+
+        for i, label in enumerate(vertex_labels):
+            if label is None:
+                vertex_labels[i] = ko
+                ko -= 1
+
+        return vertex_labels
+
+    def order_edges(G, vertex_labels):
+        n = len(vertex_labels)
+        m = 1
+        edge_labels = {}
+
+        for j in range(2, n + 1):
+            for i in range(1, j):
+                if G.has_edge(i, j):
+                    edge_labels[(i, j)] = m
+                    m += 1
+
+        return edge_labels
+
+    def is_upset_of_poset(Poset, subset, keys):
+        for (u, v) in subset:
+            for (w, x) in keys:
+                if (Poset[(u, v), (w, x)] == 1 and (w, x) not in subset):
+                    return False
+        return True
+
+    def generate_orientations(globO, starting_of_Ek, m, k, keys):
+        # Creating a poset
+        Poset = {}
+        for i in range(starting_of_Ek, m - 1):
+            for j in range(starting_of_Ek, m - 1):
+                u, v = keys[i]
+                w, x = keys[j]
+                Poset[(u, v), (w, x)] = 0
+
+        # Create a new graph to determine reachable vertices
+        new_G = DiGraph()
+        for i in range(0, starting_of_Ek):
+            u, v = keys[i]
+            if globO[(u, v)] == 1:
+                new_G.add_edge(v, u)
+            else:
+                new_G.add_edge(u, v)
+
+        for i in range(starting_of_Ek, m - 1):
```

Comment:
To avoid this for loop, it suffices to add all vertices of `G` to `new_G`, for instance using `new_G = DiGraph([G, []], format='vertices_and_edges')`.